framework/src/materials/MaterialBase.C

```diff
@@ -0,0 +1,210 @@
+    MeshChangedInterface(parameters),
+
+    // The false flag disables the automatic call buildOutputVariableHideList;
+    // for Material objects the hide lists are handled by MaterialOutputAction
+    OutputInterface(parameters, false),
+    RandomInterface(parameters, *parameters.get<FEProblem *>("_fe_problem"), parameters.get<THREAD_ID>("_tid"), false),
+    _subproblem(*parameters.get<SubProblem *>("_subproblem")),
+    _fe_problem(*parameters.get<FEProblem *>("_fe_problem")),
+    _tid(parameters.get<THREAD_ID>("_tid")),
+    _assembly(_subproblem.assembly(_tid)),
+    _bnd(_material_data_type != Moose::BLOCK_MATERIAL_DATA),
+    _neighbor(_material_data_type == Moose::NEIGHBOR_MATERIAL_DATA),
+    _qp(std::numeric_limits<unsigned int>::max()),
+    _qrule(_bnd ? _assembly.qRuleFace() : _assembly.qRule()),
+    _JxW(_bnd ? _assembly.JxWFace() : _assembly.JxW()),
+    _coord(_assembly.coordTransformation()),
+    _q_point(_bnd ? _assembly.qPointsFace() : _assembly.qPoints()),
+    _normals(_assembly.normals()),
+    _current_elem(_neighbor ? _assembly.neighbor() : _assembly.elem()),
+    _current_side(_neighbor ? _assembly.neighborSide() : _assembly.side()),
+    _mesh(_subproblem.mesh()),
+    _coord_sys(_assembly.coordSystem()),
+    _has_stateful_property(false)
+{
+  // Fill in the MooseVariable dependencies
+  const std::vector<MooseVariable *> & coupled_vars = getCoupledMooseVars();
+  for (unsigned int i=0; i<coupled_vars.size(); i++)
+    addMooseVariableDependency(coupled_vars[i]);
+}
+
+
+void
+MaterialBase::initStatefulProperties(unsigned int n_points)
+{
+  if (_has_stateful_property)
+    for (_qp = 0; _qp < n_points; ++_qp)
+      initQpStatefulProperties();
+}
+
+
+void
+MaterialBase::initQpStatefulProperties()
+{
+  mooseDoOnce(mooseWarning(std::string("Material \"") + name() + "\" declares one or more stateful properties but initQpStatefulProperties() was not overridden in the derived class."));
+}
+
+
+void
+MaterialBase::setZeroPropAsRequested(const std::string & prop_name)
+{
+  // look if property name is in _zero_props
+  std::set<std::string>::iterator it = _zero_props.find(prop_name);
+  if (it != _zero_props.end())
+  {
+    // move the property from _zero_props to _requested_props
+    _requested_props.insert(*it);
+    _zero_props.erase(it);
+  }
+}
+
+
+void
+MaterialBase::checkStatefulSanity() const
+{
+  for (std::map<std::string, int>::const_iterator it = _props_to_flags.begin(); it != _props_to_flags.end(); ++it)
+  {
+    if (static_cast<int>(it->second) % 2 == 0) // Only Stateful properties declared!
+      mooseError("Material '" << name() << "' has stateful properties declared but not associated \"current\" properties." << it->second);
+  }
+}
+
+
+void
+MaterialBase::registerPropName(std::string prop_name, bool is_get, MaterialBase::Prop_State state)
+{
+  if (!is_get)
+  {
+    _props_to_flags[prop_name] |= static_cast<int>(state);
+    if (static_cast<int>(state) % 2 == 0)
```

Comment:
I guess this is old code (just moved) but somebody went through the trouble of using a bitfield here and even "or'd" things together but then did a mathematical operator instead of a bit operator on this line. It might have been me though so I guess I won't poke too much fun.